src/webui/webui.h

```diff
@@ -61,17 +68,31 @@ class WebUI final : public ApplicationComponent<QObject>
     QHostAddress hostAddress() const;
     quint16 port() const;
 
+    QString username() const;
+    nonstd::expected<void, QString> setUsername(const QString &username);
+    nonstd::expected<void, QString> setPassword(const QString &password);
+    QString tempPassword() const;
+
 signals:
     void error(const QString &message);
 
 private slots:
     void configure();
 
 private:
+    void setError(QStringView message);
+
+    SettingValue<QString> m_storeUsername;
+    SettingValue<QByteArray> m_storePasswordHash;
+
     bool m_isEnabled = false;
     bool m_isErrored = false;
     QString m_errorMsg;
     QPointer<Http::Server> m_httpServer;
     QPointer<Net::DNSUpdater> m_dnsUpdater;
     QPointer<WebApplication> m_webapp;
+
+    bool m_isTempPasswordAllowed = false;
+    QString m_tempPassword;
```

Comment:
It is a red flag that you keep the clear text password in memory. You should avoid doing so and use other means.
Also ideally this field should be allocated on the stack instead of on the heap.
Note that securely erasing a string isn't even easy! https://stackoverflow.com/questions/5698002/how-does-one-securely-clear-stdstring

Reply:
> It is a red flag that you keep the clear text password in memory. You should avoid doing so and use other means.

Do you really believe that we should take care of those cases when someone malicious has sufficient access to a computer to read its memory? How can we counteract it in any way effectively?

Reply:
>Do you really believe that we should take care of those cases when someone malicious has sufficient access to a computer to read its memory? How can we counteract it in any way effectively?

Yes. It is not local access to the computer. It could be remote access with out-of-bound memory read/write.